File: backend/app/schemas/Activity.py
import uuid
from pydantic import BaseModel, EmailStr
from typing import List, Optional
from datetime import datetime


class ActivityCreate(BaseModel):
    user_id: uuid.UUID
    title: str
    description: Optional[str] = None
    tags: Optional[List[str]] = []
    count: Optional[int] = 0

class ActivityOut(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    description: Optional[str]
    tags: List[str]
    count: int
    created_at: Optional[datetime]
    updated_at: Optional[datetime]

File: frontend/src/api/activity.ts
import Api from "./Api";
import type { AxiosError } from "axios";
import type { EnergyLevel } from "../types/EnergyLevel";

export interface ActivityTemplateOut {
  id: string; // UUID
  title: string;
  description?: string;
  duration_minutes?: number;
  good_point?: string;
  insight?: string;
  created_at: string;
  updated_at: string;
  energy_level: EnergyLevel; // 0~10 숫자 타입
}

export interface ActivityCreatePayload {
    user_id : string;
    title : string;
    description : string;
    is_public : false;
    duration_minutes? : number;
    good_point? : string;
    insight? : string;
    energy_level: EnergyLevel; // 0~10 숫자 타입
}

// class ActivityCreate(BaseModel):
//     user_id: UUID
//     title: str
//     description: Optional[str] = None
//     is_public: Optional[bool] = False
//     duration_minutes: Optional[int] = None
//     good_point: Optional[str] = None
//     insight: Optional[str] = None
//         energy_level: EnergyLevelEnum  

/*----------------------------------------------
 * ✅ ActivityTemplate 리스트 가져오기
 ----------------------------------------------*/
export async function fetchActivityTemplates(): Promise<ActivityTemplateOut[]> {
  try {
    const res = await Api.get("/activities/templates");
    return res.data;
  } catch (error) {
    const axiosError = error as AxiosError<{ detail?: string }>;
    const msg = axiosError.response?.data?.detail || "ActivityTemplate 조회 실패";
    throw new Error(msg);
  }
}
